equinox/nn/_spectral_norm.py

```diff
@@ -1,18 +1,19 @@
-from typing import Generic, Optional, TypeVar
+from typing import Callable, Generic, Optional, TypeVar
 
 import jax
 import jax.lax as lax
 import jax.numpy as jnp
 import jax.random as jr
 from jaxtyping import Array, Float, PRNGKeyArray
 
+from .._eval_shape import filter_eval_shape
 from .._module import field
 from .._tree import tree_at
 from ._sequential import StatefulLayer
 from ._stateful import State, StateIndex
 
 
-def _power_iteration(weight, u, v, eps):
+def _power_iteration_old(weight, u, v, eps):
```

Comment:
This function should be removeable? We can implement it in terms of the new one.